src/js/integrations/stalltracking.ts

```diff
@@ -0,0 +1,328 @@
+/* eslint-disable max-lines */
```

Comment:
This is a nice approach - will do a more detailed review in a bit. What do you think we need to do to move this into sentry-javascript.

Also the monkey patching is not ideal, we should update tracing to make it easier to patch rather than patch the idle transaction this way tbh.

Reply:
Yeah I strongly agree the monkey patching is bad, but we reckoned that we're going to release this as a prerelease to see how it goes, if it's actually effective, and if it's valuable to customers, if it turns out well we can then move it over to the javascript repo and solidify it by updating tracing to actually support hooks to the start and finish methods.

Reply:
Sounds like a plan 👍